enterprise/models.py

```diff
@@ -2,42 +2,235 @@
+    EnterpriseCustomerHistory is an archive model for EnterpriseCustomer.
+
+    Note that `members` field is omitted to avoid storing huge amounts of user affinity data.
+
+    Fields:
+        id (IntegerField): Enterprise Customer identified
+        name (CharField): Enterprise Customer name
+        short_code (CharField): Enterprise Customer code - used to reference this Enterprise Customer in other
+                                    parts of the system (SSO, ecommerce, analytics etc.)
+        logo (ImageField):  Enterprise Customer logo
+        data_sharing_policy (CharField): stores data sharing policy: either no data sharing,
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
     """
 
-    # TODO: add field definitions
+    class Meta(object):
+        """
+        Meta class for EnterpriseCustomerHistory model.
+        """
+
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer History"
+        verbose_name_plural = "Enterprise Customer Histories"
+
+    origin = models.ForeignKey(EnterpriseCustomer, related_name="history")
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_("Enterprise Customer name."))
+    short_code = models.CharField(
+        max_length=120, blank=False, null=False,
+        help_text=_("Short code will be used to reference this Enterprise Customer in other parts of the system,"
+                    "i.e. SSO, ecommerce app, analytics etc.")
+    )
+    logo = models.ImageField(
+        upload_to="enterprise_logos",
+        help_text=_("Please add only .PNG files for logo images."),
+        null=True, blank=True, max_length=255
+    )
+    data_sharing_policy = models.CharField(
+        blank=False, null=False, help_text=_("Data sharing policy"), max_length=20,
+        choices=(
+            (EnterpriseCustomer.NO_DATA_SHARING, _("No data sharing requested")),
+            (EnterpriseCustomer.REQUEST_DATA_SHARING, _("Request data sharing")),
+            (EnterpriseCustomer.REQUIRE_DATA_SHARING, _("Require data sharing")),
+        )
+    )
+    active = models.BooleanField(default=True)
 
     def __str__(self):
         """
-        Get a string representation of this model instance.
+        Return human-readable string representation.
+        """
+        return "<EnterpriseCustomerHistory {ID}: {name}, Last Modified: {modified}>".format(
+            ID=self.id,
+            modified=self.modified,
+            name=self.name,
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class EnterpriseCustomerMember(models.Model):
```

Comment:
Unfortunately "member" is an overloaded term at edX, and it will be difficult to change the existing connotation.  So this model should probably be renamed to `EnterpriseCustomerUser`, which is a better description of what the data actually is, anyway.